conformance/lang/expressions/let-expr/let_expr.balt

```diff
@@ -0,0 +1,600 @@
+    io:println(let boolean b1 = true, boolean b2 = false in {"b": b1, b1: b1, b2, ["a"]: [b1, b2]}); // @output {"b":true,"b1":true,"b2":false,"a":[true,false]}
+    io:println(let boolean b1 = true, boolean b2 = false in {"b": {b1, b2}}); // @output {"b":{"b1":true,"b2":false}}
+}
+
+Test-Case: output
+Description: Test let-expr when expression after `in` is a mapping-constructor-expr.
+Labels: let-expr, nil-literal, nil-type, record-type, union-type, int, function-call-expr, mapping-constructor-expr
+
+type Record record {|
+    int|boolean? i;
+|};
+
+function init() {
+    Record r = let null n = () in {i: n};
+    io:println(r); // @output {"i":null}
+}
+
+Test-Case: output
+Description: Test let-expr when expression after `in` is a mapping-constructor-expr.
+Labels: let-expr, boolean, boolean-literal, union-type, record-type, int, function-call-expr, mapping-constructor-expr
+
+type Record record {|
+    int|boolean? i;
+|};
+
+function init() {
+    Record r = let boolean b = false in {i: b};
+    io:println(r); // @output {"i":false}
+    io:println(let boolean b1 = true, boolean b2 = !b1 in {"b": {b1, b2}}); // @output {"b":{"b1":true,"b2":false}}
+}
+
+Test-Case: output
+Description: Test let-expr when expression after `in` is a mapping-constructor-expr.
+Labels: let-expr, string, record-type, array-type, mapping-constructor-expr
+
+type Names record {|
+    string country;
+    string[] names;
+|};
+
+function init() {
+    record {| string myField; |} r = let string smiley = "h😀llo" in {myField: smiley};
+    io:println(r.myField); // @output h😀llo
+}
+
+Test-Case: output
+Description: Test let-expr when expression after `in` is a table-constructor-expr.
+Labels: let-expr, string, boolean, boolean-literal, record-type, array-type, function-call-expr, table-constructor-expr
+
+type Names record {|
+    string country;
+    string[] names;
+|};
+
+function init() {
+    io:println(let null n = () in table key(n1) [{n1: "N1", n2: n}, {n1: "N2", n2: n}]); // @output [{"n1":"N1","n2":null},{"n1":"N2","n2":null}]
+    io:println(let boolean b1 = true, boolean b2 = true
+        in table key(name) [{isValid: b1, name: "XX"}, {isValid: b2, name: "YY"}]); // @output [{"isValid":true,"name":"XX"},{"isValid":true,"name":"YY"}]
+    io:println(let boolean b1 = true, boolean b2 = false in table key(name) [{isValid: b1 && b2, name: "XX"}]); // @output [{"isValid":false,"name":"XX"}]
+    table<Names> tbNames = table [];
+    _ = let Names val = let string[] names = ["Sam", "John", "Ann"] in {country: "Ireland", names: names} in tbNames.add(val);
+    io:println(tbNames); // @output [{"country":"Ireland","names":["Sam","John","Ann"]}]
+}
+
+Test-Case: output
+Description: Test let-expr when expression after `in` is an object-constructor-expr.
+Labels: let-expr, nil-literal, nil-type, function-call-expr, object-type, object-constructor-expr, equality,
+        method-call-expr
```

Comment:
labels: module-class-defn, var, field-access-expr. Check other places we have class-defn

Reply:
Added